### PR TITLE
Closes #1458: webp compatibility

### LINKED ISSUE #1458: Media Webp support
It would be good if wp-rocket supports this type of plugins, to be able to use Webp (Other cache plugins already support it)
Wp-rocket should detect when the browser is compatible and load the image in Webp format
Plugin: https://wordpress.org/plugins/media-webp/

A plugin that does this: https://wordpress.org/plugins/cache-enabler/

### PR DESCRIPTION
**Pull request by Screenfeed**
Returning `true` to the filter `'rocket_disable_webp_cache'` provides a quick way to disable webp cache.
For more advanced management, like adding compatibility with a third party webp plugin, an instance using the interface `WP_Rocket\Subscriber\Third_Party\Plugins\Images\Webp\Webp_Interface` must be added to the array filtered by `'rocket_webp_plugins'`.

### CODE REVIEW

File: inc/classes/subscriber/Media/class-webp-subscriber.php [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+	 * @since  3.4
+	 * @access public
+	 * @author Remy Perona
+	 * @author Grégory Viguier
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	public function convert_to_webp( $html ) {
+		if ( ! $this->options->get( 'cache_webp' ) || $this->get_plugins_serving_webp() ) {
```
Comment (Tabrisrp) [inline inc/classes/subscriber/Media/class-webp-subscriber.php]: Shouldn't it be `! $this->get_plugins_serving_webp()`?

File: inc/classes/subscriber/third-party/plugins/Images/Webp/class-shortpixel-subscriber.php [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+	private $options;
+
+	/**
+	 * CDNSubscriber instance.
+	 *
+	 * @var    CDNSubscriber
+	 * @access private
+	 * @author Grégory Viguier
+	 */
+	private $cdn_subscriber;
```
Comment (Tabrisrp) [inline inc/classes/subscriber/third-party/plugins/Images/Webp/class-shortpixel-subscriber.php]:
```suggestion
	private $cdn;
```

File: inc/classes/subscriber/third-party/plugins/Images/Webp/class-shortpixel-subscriber.php [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+	 * @var    Options_Data
+	 * @access private
+	 * @author Remy Perona
+	 */
+	private $options;
+
+	/**
+	 * CDNSubscriber instance.
+	 *
+	 * @var    CDNSubscriber
```
Comment (Tabrisrp) [inline inc/classes/subscriber/third-party/plugins/Images/Webp/class-shortpixel-subscriber.php]:
```suggestion
	 * @var    CDN
```

File: inc/classes/subscriber/third-party/plugins/Images/Webp/class-shortpixel-subscriber.php [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+	 * Options_Data instance.
+	 *
+	 * @var    Options_Data
+	 * @access private
+	 * @author Remy Perona
+	 */
+	private $options;
+
+	/**
+	 * CDNSubscriber instance.
```
Comment (Tabrisrp) [inline inc/classes/subscriber/third-party/plugins/Images/Webp/class-shortpixel-subscriber.php]:
```suggestion
	 * CDN instance.
```

File: inc/classes/subscriber/Media/class-webp-subscriber.php [hunk 5]
```diff
@@ -0,0 +1,593 @@
+<?php
+namespace WP_Rocket\Subscriber\Media;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Subscriber\CDN\CDNSubscriber;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\Subscriber\Third_Party\Plugins\Images\Webp\Webp_Interface;
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/subscriber/Media/class-webp-subscriber.php]: Only in `get_webp_plugins()` but this `use` is indeed useless.

[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/subscriber/Media/class-webp-subscriber.php]: Well, the `use` is only useful for comments, so we may consider removing it.

File: inc/classes/subscriber/Media/class-webp-subscriber.php (hunk 1 above)
Comment (Screenfeed) [reply inc/classes/subscriber/Media/class-webp-subscriber.php]: No.
If a plugin is serving webp, then we don't need to do anything.

**Pull request merged**